homeassistant/components/decora_wifi/__init__.py

```diff
@@ -1 +1,135 @@
+
+from .common import CommFailed, DecoraWifiPlatform, LoginFailed, LoginMismatch
+from .const import CONF_OPTIONS, CONF_TITLE, DEFAULT_SCAN_INTERVAL, DOMAIN, PLATFORMS
+
+_LOGGER = logging.getLogger(__name__)
+
+DECORAWIFI_HOST_SCHEMA = vol.Schema({vol.Required(CONF_HOST): cv.string})
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): cv.string,
+                    vol.Required(CONF_PASSWORD): cv.string,
+                    vol.Optional(CONF_SCAN_INTERVAL, default=120): cv.positive_int,
+                },
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the Decora Wifi component."""
+    hass.data.setdefault(
+        DOMAIN, EntityComponent(_LOGGER, DOMAIN, hass, timedelta(DEFAULT_SCAN_INTERVAL))
+    )
+
+    if DOMAIN not in config:
+        return True
+
+    # Trigger import config flow
+    conf = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={
+                    CONF_USERNAME: conf[CONF_USERNAME],
+                    CONF_PASSWORD: conf[CONF_PASSWORD],
+                },
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry):
+    """Set up Decora Wifi from a config entry."""
+
+    conf_data = dict(entry.data)
+    email = conf_data[CONF_USERNAME]
+    password = conf_data[CONF_PASSWORD]
+
+    component: EntityComponent = hass.data[DOMAIN]
+
+    # Set a sane default scan interval.
+    conf_data[CONF_OPTIONS] = {
+        CONF_SCAN_INTERVAL: dict(entry.options).get(
+            CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+        )
+    }
+
+    try:
+        # Attempt to log in.
+        session = await DecoraWifiPlatform.async_setup_decora_wifi(
+            hass,
+            email,
+            password,
+        )
+    except LoginFailed as exc:
+        _LOGGER.error("Login failed")
+        raise ConfigEntryAuthFailed from exc
+    except CommFailed as exc:
+        _LOGGER.error("Communication with myLeviton failed")
+        raise ConfigEntryNotReady from exc
+
+    # Verify the Unique ID matches, then add the entity to hass
+    if session.unique_id == conf_data[CONF_ID]:
+        await component.async_add_entities([session], False)
+    else:
+        _LOGGER.error("Userid mismatch with config entry")
+        raise LoginMismatch("Userid mismatch with config entry")
+    conf_data[CONF_ENTITY_ID] = session.entity_id
+    hass.config_entries.async_update_entry(
+        entry, title=f"{CONF_TITLE} - {CONF_USERNAME}", data=conf_data
+    )
+
+    activeplatforms = session.active_platforms
```

Comment:
```suggestion
    active_platforms = session.active_platforms
```

Please use snake case